google/cloud/forseti/common/gcp_api/repository_mixins.py

```diff
@@ -447,14 +447,67 @@ def update(self, resource, data, target=None, verb='update', **kwargs):
 
         if self.read_only:
             LOGGER.info(
-                'Update called on a read only repository, no action taken on '
-                'target %s, resource %s for data %s.', target, resource, data)
+                '%s called on a read only repository, no action taken on '
+                'target %s, resource %s for data %s.', verb, target, resource,
+                data)
             resource_link = self._build_resource_link(**arguments)
             return _create_fake_operation(resource_link, verb, target)
 
         return self.execute_command(verb=verb, verb_arguments=arguments)
 
 
+class PatchResourceMixin(_ModifyResourceBaseMixin):
+    """Mixin that implements the Patch API command for resources."""
+
+    def patch(self, resource, data, target=None, verb='patch', **kwargs):
```

Comment:
if this is for patch only, why is verb an argument instead of a fixed value? (same for update)

Reply:
Good question, the main reason is that some APIs may use a different verb for the same meaning, so this gives flexibility when using the mixin.